modules: hal_infineon: updated CMakeLists.txt for whd-expansion

violatedasyntax opened the pull request:
Updated the CMakeLists.txt to use bins directly instead of C files with whd-expansion.
Depends on: https://github.com/zephyrproject-rtos/hal_infineon/pull/31

Signed-off-by: Chaitanya Gaikwad ChaitanyaSandeep.Gaikwad@infineon.com

sreeramIfx commented on drivers/wifi/infineon/Kconfig.airoc:
```diff
@@ -254,6 +254,11 @@ config CYW955573M2IPA1_SM
 	help
 	  Infineon CYW955573M2IPA1 (SM) module
+
+config CYW955513UBG_SM
```
```suggestion
config CYW55513IUBG_SM
```
Was this the intention?

violatedasyntax committed:
drivers: wifi: Infineon whd-expansion 1.2.1

Updated CMakeLists.txt to use bins directly with whd-expansion.
Updated Kconfig.airoc to include CYW55513UBG_SM.
Updated west.yml to latest hal_infineon revision.

Signed-off-by: Chaitanya Gaikwad <ChaitanyaSandeep.Gaikwad@infineon.com>
